src/components/ServiceEndpoints/ServiceEndpoint.tsx

```diff
@@ -1,54 +1,93 @@
-import type { DidUri } from '@kiltprotocol/sdk-js';
+import {
+  Credential,
+  DidUri,
+  ICredential,
+  KiltPublishedCredentialCollectionV1,
+  KiltPublishedCredentialCollectionV1Type,
+} from '@kiltprotocol/sdk-js';
 
-import { useCallback, useState } from 'react';
+import { Fragment, useCallback, useState } from 'react';
 import classnames from 'classnames';
 
+import { map } from 'lodash-es';
+
 import * as styles from './ServiceEndpoint.module.css';
 
 import { CredentialVerifier } from '../Credential/Credential';
 import { useBooleanState } from '../../hooks/useBooleanState';
 
+function isLegacyCredential(credential: unknown): credential is {
+  request: ICredential;
+} {
+  return (
+    typeof credential === 'object' &&
+    credential !== null &&
+    'request' in credential
+  );
+}
+
+function isPublishedCollection(
+  json: unknown,
+  endpointType: string,
+): json is KiltPublishedCredentialCollectionV1 {
+  if (endpointType !== KiltPublishedCredentialCollectionV1Type) {
+    return false;
+  }
+  if (!Array.isArray(json)) {
+    return false;
+  }
+  const [{ credential }] = json as KiltPublishedCredentialCollectionV1;
+  return Credential.isICredential(credential);
+}
+
 interface Props {
   url: string;
   endpointType: string;
   did: DidUri;
 }
 
 export function ServiceEndpoint({ url, endpointType, did }: Props) {
-  const [credential, setCredential] = useState();
+  const [credentials, setCredentials] = useState<ICredential[]>();
 
   const fetching = useBooleanState();
   const fetched = useBooleanState();
-  const [error, setError] = useState<string>();
+  const error = useBooleanState();
 
-  const handleFetch = useCallback(async () => {
-    if (fetched.current) {
-      fetched.off();
-      setCredential(undefined);
-      return;
-    }
+  const handleClose = useCallback(() => {
+    error.off();
+    fetched.off();
+    setCredentials(undefined);
+  }, [error, fetched]);
 
+  const handleFetch = useCallback(async () => {
     try {
-      if (credential) {
+      fetching.on();
+      const response = await fetch(url);
+      const json = await response.json();
+
+      if (isLegacyCredential(json)) {
+        setCredentials([json.request]);
         return;
       }
 
-      fetching.on();
-      const response = await fetch(url);
-      const result = await response.json();
-      setCredential(result);
+      if (Credential.isICredential(json)) {
+        setCredentials([json]);
+        return;
+      }
 
-      if (!did) {
-        throw new Error('No DID');
+      if (isPublishedCollection(json, endpointType)) {
+        setCredentials(map(json, 'credential'));
+        return;
       }
-    } catch (error) {
-      console.log(error);
-      setError('Cannot fetch the credentials from the given endpoint');
+
+      throw new Error();
+    } catch (err) {
```

Comment:
Is this `err` needed?
```suggestion
    } catch {
```